src/profilers.cpp

```diff
@@ -108,15 +155,13 @@ namespace aperf {
                 "and metric-read in \"" + this->get_name() + "\"! Terminating "
                 "the profiled command wrapper.", true, true);
           kill(pid, SIGTERM);
-          return;
+          return ERROR_PIPE_METRIC_EXEC;
```

Comment:
Since we return ```ERROR_PIPE_METRIC_EXEC``` in case of the failure of the pipe syscall, I'd move the handling code (i.e. printing the error message and terminating the wrapper) to ```this->process```.

Reply:
Done :+1:

Reply:
Nice, I'd also remove ```kill(pid, SIGTERM)``` from here as this is already done by the error handling code later.